### PR TITLE
LPS-174539 Add 'modifiable' Object Definition property

### PR DESCRIPTION
**Pull request by mateussgn**
Related issue: https://issues.liferay.com/browse/LPS-174539

### CODE REVIEW

**Comment by mateussgn** [comment]
I checked the CI results and the failures are unrelated.

File: modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionVulcanBatchEngineTaskItemDelegateTest.java [hunk 1]
```diff
@@ -182,6 +182,7 @@ private ObjectDefinition _createObjectDefinition(String name) {
 					RandomTestUtil.randomString());
 				id = RandomTestUtil.randomLong();
 				label = Collections.singletonMap("en_US", "O" + sanitizedName);
+				modifiable = true;
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionVulcanBatchEngineTaskItemDelegateTest.java]: Can you change this logic in order to have the value true just when the object definition is a custom object?

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java [hunk 2]
```diff
@@ -657,7 +657,13 @@ private ObjectDefinition _toObjectDefinition(
 				id = objectDefinition.getObjectDefinitionId();
 				label = LocalizedMapUtil.getLanguageIdMap(
 					objectDefinition.getLabelMap());
-				modifiable = objectDefinition.getModifiable();
+
+				if (!GetterUtil.getBoolean(
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java:661]: I think this condition is wrong, it must validate if the feature flag is active/true

File: modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionVulcanBatchEngineTaskItemDelegateTest.java [hunk 3]
```diff
@@ -182,7 +182,13 @@ private ObjectDefinition _createObjectDefinition(String name) {
 					RandomTestUtil.randomString());
 				id = RandomTestUtil.randomLong();
 				label = Collections.singletonMap("en_US", "O" + sanitizedName);
-				modifiable = true;
+
+				if (!GetterUtil.getBoolean(
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionVulcanBatchEngineTaskItemDelegateTest.java:186]: I think this condition is wrong, it must validate if the feature flag is active/true

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:object

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by evanilson-santana** [comment]
ci:test:object

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java [hunk 4]
```diff
@@ -296,6 +303,12 @@ public ObjectDefinition putObjectDefinition(
 
 		// TODO Move logic to service
 
+		if (Validator.isNull(objectDefinition.getModifiable()) &&
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java:306]: I think this condition is wrong, did you have a chance to test it through REST API?

File: modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java [hunk 5]
```diff
@@ -223,6 +224,12 @@ public ObjectDefinition postObjectDefinition(
 			ObjectDefinition objectDefinition)
 		throws Exception {
 
+		if (Validator.isNull(objectDefinition.getModifiable()) &&
```
Comment (guilhermedcamacho) [inline modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java:227]: I think this condition is wrong, did you have a chance to test it through REST API?

**Review by guilhermedcamacho — CHANGES_REQUESTED** [review]

**Comment by mateussgn** [comment]
ci:close

**Pull request closed without merging**